packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1121,8 +1121,11 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   AnimationController _floatingCursorResetController;
 
+  // True iff the text has ever changed since instantiation.
+  bool _textChanged = false;
+
   @override
-  bool get wantKeepAlive => widget.focusNode.hasFocus;
+  bool get wantKeepAlive => widget.focusNode.hasFocus || _textChanged;
```

Comment:
@goderbauer This was the first thing that came to my mind, but I couldn't find a better way after digging into it a bit. Let me know if there's something more robust that you're aware of.